addons/sale_product_configurator/views/templates.xml

```diff
@@ -32,6 +32,10 @@
 
             <input type="hidden" class="product_template_id" t-att-value="product.id"/>
             <input type="hidden" class="product_id" t-attf-name="product_id" t-att-value="product_variant.id"/>
+            <input type="hidden"
+                   class="has_optional_products"
+                   t-attf-name="has_optional_products"
+                   t-att-value="product_variant.optional_product_ids.filtered(lambda p: p._is_add_to_cart_possible(combination))"/>
```

Comment:
```suggestion
            <input type="hidden" class="has_optional_products" t-attf-name="has_optional_products"
                   t-att-value="product_variant.optional_product_ids.filtered(lambda p: p._is_add_to_cart_possible(combination))"/>
```
it is nice to split line by attribute (depend on who you ask tho :yum:), but don't "over-do" it, group at least a few attribute on the same line, or at least don't do one attribute per line when your attribute is 10 char